src/sentry/notifications/manager.py

```diff
@@ -71,74 +33,57 @@ def get_settings(
         self,
         provider: ExternalProviders,
         type: NotificationSettingTypes,
-        user=None,
-        team=None,
-        project=None,
-        organization=None,
-    ):
+        user: Optional = None,
+        team: Optional = None,
+        project: Optional = None,
+        organization: Optional = None,
+    ) -> NotificationSettingOptionValues:
         """
         In this temporary implementation, always read EMAIL settings from
         UserOptions. One and only one of (user, team, project, or organization)
         must not be null. This function automatically translates a missing DB
         row to NotificationSettingOptionValues.DEFAULT.
-        :param provider: ExternalProviders enum
-        :param type: NotificationSetting.type enum
-        :param user: (optional) A User object
-        :param team: (optional) A Team object
-        :param project: (optional) A Project object
-        :param organization: (optional) An Organization object
-        :return: NotificationSettingOptionValues enum
         """
-        user_id_option = getattr(user, "id", None)
-        scope_type, scope_identifier = _get_scope(
-            user_id_option, project=project, organization=organization
-        )
-        target = _get_target(user, team)
-
-        value = (  # NOQA
-            self.filter(
-                provider=provider.value,
-                type=type.value,
-                scope_type=scope_type,
-                scope_identifier=scope_identifier,
-                target=target,
-            ).first()
-            or NotificationSettingOptionValues.DEFAULT
+        from sentry.models import UserOption
+
+        setting_option = self.find_settings(
+            provider, type, user, team, project, organization
+        ).first()
```

Comment:
Careful about using `.first()`. If your queryset doesn't have a sort order specified, it adds one in, which can break index use. If all you want is the first setting, it can be safer to use `list(qs)[0]`

Reply:
The `unique_together` constraint should make it impossible to get more than one value, but I don't want to accidentally make the query slower. I'll update it and add a comment.